push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/serviceLease/ServiceDisposalScheduler.java

```diff
@@ -0,0 +1,70 @@
+package org.jboss.aerogear.unifiedpush.message.serviceLease;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+
+/**
+ * Allows to scheduled instantiated services for disposal.
+ *
+ * Gracefully disposes services on a container shutdown.
+ */
+@Singleton
+@Startup
+public class ServiceDisposalScheduler {
+
+    private ScheduledExecutorService scheduler;
+    private long terminationTimeout = 10000L;
+
+    /**
+     * Creates a new scheduler on container startup
+     */
+    @PostConstruct
+    public void initialize() {
+        scheduler = Executors.newSingleThreadScheduledExecutor();
```

Comment:
@matzew in https://github.com/lfryc/aerogear-unifiedpush-server/pull/16#discussion_r56665401

> we could use java ee concurrency ?